### DISCUSSION TITLE
Some Suggestions For Future Iteration

### DISCUSSION THREAD

**Discussion by TaxicabMessiah** [zaps166/QMPlay2#784] [Ideas]
This is the only media player I use now and have some more input if desired.
-Inability to highlight multiple media files and group edit metadata
-Cannot drag and drop image files into the metadata window
-No waveform add-on of active media file available so we can see if it's brickwalled or just observe swings in volume
-Quirks with background image.  Won't retain detail of a 4k image, seems to compress the image before displaying.  Has no "shrink to fit" or "center" type options like Windows wallpaper would.

Thanks for reading!

**Comment by zaps166** [comment]
> Quirks with background image. Won't retain detail of a 4k image, seems to compress the image before displaying. Has no "shrink to fit" or "center" type options like Windows wallpaper would.

I know. Isn't it enough to do it manually once? :sweat_smile: How often do you set wallpaper? Anyway it'd be better to select a file and perform scaling on load instead of copying full image data into text configuration file (IIRC).

> No waveform add-on of active media file available so we can see if it's brickwalled or just observe swings in volume

(If I remember your point correctly) It's displayed in real-time only. No file analysis is performed. Currently to much to do for this feature.

>  Cannot drag and drop image files into the metadata window

Could you explain?

> Inability to highlight multiple media files and group edit metadata

It can be a feature request, but not going to be done soon (not quick change in software). How do you see this feature?

**Reply by TaxicabMessiah** [reply]
I must have changed some settings because now I'm getting full resolution on wallpaper but also getting the image metadata from online radio to display in the new version of this player.  Pretty cool.
![Screenshot 2024-12-11 001113](https://github.com/user-attachments/assets/adceedaf-9803-4747-9ff7-f0db360df8d2)

**Reply by zaps166** [reply]
> I must have changed some settings because now I'm getting full resolution on wallpaper

Maybe because you switched your package to Qt6 (newer version of tool I use)?

> but also getting the image metadata from online radio to display in the new version of this player. Pretty cool.

Yes, QMPlay2 24.06.16 and 24.05.23 had a bug which I fixed (#770).

**Comment by TaxicabMessiah** [comment]
-This is an example in Foobar2000 of what I mean by displaying the wave form of active media (image with the red arrow)
-This is me trying to drag and drop an image file into the properties/metadata window in QMPlay2.  It's rejected, you have to hit the "upload" button and do it manually. (image with yellow arrow)

![Screenshot 2024-12-08 033611](https://github.com/user-attachments/assets/89645d8c-5086-4765-b742-891221b45db5)


![Screenshot 2024-12-08 033141](https://github.com/user-attachments/assets/3429faf7-568e-42b7-9374-9d5f624c54fa)

**Reply by zaps166** [reply]
>  It's rejected, you have to hit the "upload" button and do it manually. (image with yellow arrow)

Aaa that should be simple to do.

> This is an example in Foobar2000 of what I mean by displaying the wave form of active media (image with the red arrow)

As I thought. Whole file analyze is needed + display a graph. It's more like audio editor than a player. I'm aware it's helpful, but it's a lot of work to do it correctly (also must be optional, disabled by default).

**Reply by TaxicabMessiah** [reply]
Right I know.  But one of the helpful features in Foobar2000 is that I can highlight 30 music files and then open up the metadata properties and just drop an image right in directly.  I'm not saying what you have is bad I'm just suggesting there are no big changes to make to this player, it's really good.  But these are the small things I've noticed using it every day now.  I agree about the waveform that it should be an optional add on like the equalizer, etc.

**Reply by zaps166** [reply]
> It's rejected, you have to hit the "upload" button and do it manually. (image with yellow arrow)

Done.

> Right I know. But one of the helpful features in Foobar2000 is that I can highlight 30 music files and then open up the metadata properties and just drop an image right in directly. I'm not saying what you have is bad I'm just suggesting there are no big changes to make to this player, it's really good. But these are the small things I've noticed using it every day now. I agree about the waveform that it should be an optional add on like the equalizer, etc.

Agree it's useful, but too much work for this for now.
